scripts/build/builders/android.py

```diff
@@ -121,11 +121,12 @@ def _build(self):
 
         # JNILibs will be copied as long as they reside in src/main/jniLibs/ABI:
         #    https://developer.android.com/studio/projects/gradle-external-native-builds#jniLibs
+        # to avoid redefined in IDE mode, copy to another place and add that path in build.gradle
 
         # We do NOT use python builtins for copy, so that the 'execution commands' are available
         # when using dry run.
         jnilibs_dir = os.path.join(
-            self.root, 'src/android/CHIPTool/app/src/main/jniLibs', self.board.AbiName())
+            self.root, 'src/android/CHIPTool/app/libs/jniLibs', self.board.AbiName())
```

Comment:
Should this build not be changed to use the cmakelists method instead?

Reply:
By default android studio will builds all so files at app/src/main/jniLibs, while in cmakelists/IDE mode, these files will also be built, so there will be a redefine error if we 1st build with copy mode, than 2nd build with cmakelists/IDE mode.

I change is path to app/libs/jniLibs, so Android Studio will not auto include them, and I added below code to include them in copy mode ( and not do in cmakelists/IDE mode) 

    if (rootProject.ext.matterBuildSrcDir == null) {
        sourceSets {
            main {
                jniLibs.srcDirs = ['libs/jniLibs']
            }
        }
    }